### PR TITLE
Add official Docker support

### PR DESCRIPTION
**Pull request by bkuhl**
It'd be great if this tool was available as a Docker container.  Depending on the npm release process, `@latest` may not be the best route.  DockerHub builds can be configured to initiate a build whenever a new `tag` or release is added to the repo.

### CODE REVIEW

**Comment by bkuhl** [comment]
This PR in particular requires that either Travis-CI be updated to build/push the docker container, or that DockerHub support be added to release a new version automatically.

**Comment by jamesramsay** [comment]
This is great. I'll checkout the DockerHub settings tomorrow and get it merged. Thanks @bkuhl

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM mhart/alpine-node
```
Comment (jamesramsay) [inline Dockerfile]: I'd prefer to use the latest LTS release and the smaller base image (excludes NPM), which is currently `mhart/alpine-node:base-6`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+RUN npm install -g hercule@latest
```
Comment (jamesramsay) [inline Dockerfile]: Above I've suggested the `mhart/alpine-node:base` which excludes NPM. This image is smaller and skips the `npm install` step, but we'll need to run the install process before building the image.

**Comment by bkuhl** [comment]
Now uses base image `mhart/alpine-node:base-6`

**Comment by jamesramsay** [comment]
@bkuhl it looks like you missed my other comment about trying to avoid including NPM in the docker image. I've investigated further and it looks like this is a more complicated route, so let's get this merged with the current approach. :shipit: 

Please could you:

- revert change the base image to `mhart/alpine-node:6`
- remove the `"-h"` from the command. It looks like this will get in the way of users trying to run commands.
- rebase on master and squash into a single commit with a commit message beginning `Build: your commit message`

I've created #392 so that I look into a more robust tagged docker images. Thanks!

**Comment by bkuhl** [comment]
Done

**Comment by jamesramsay** [comment]
Thanks @bkuhl 👍 

Docker image is now live: https://hub.docker.com/r/jamesramsay/hercule

**Comment by bkuhl** [comment]
Thanks!

**Pull request merged**